tools/twix/src/panels/manual_crossing_marker.rs

```diff
@@ -0,0 +1,284 @@
+            self.crossings.left_penalty_field,
+            self.crossings.left_penalty_corner,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.left_penalty_field,
+            self.crossings.left_goal_field,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.left_goal_field,
+            self.crossings.left_goal_corner,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.left_goal_field,
+            self.crossings.right_goal_field,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.left_penalty_corner,
+            self.crossings.right_penalty_corner,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.left_goal_corner,
+            self.crossings.right_goal_corner,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.right_goal_corner,
+            self.crossings.right_goal_field,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.right_penalty_corner,
+            self.crossings.right_penalty_field,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.right_goal_field,
+            self.crossings.right_penalty_field,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+        if let (Some(start), Some(end)) = (
+            self.crossings.right_penalty_field,
+            self.crossings.right_field_corner,
+        ) {
+            self.show_line_segment(&painter, start, end);
+        }
+
+        if let Some(position) = image_response.hover_pos() {
+            let position = painter.transform_pixel_to_world(position);
+            painter.line_segment(
+                point![position.x, 0.0],
+                point![position.x, 480.0],
+                Stroke::new(1.0, Color32::WHITE),
+            );
+            painter.line_segment(
+                point![0.0, position.y],
+                point![640.0, position.y],
+                Stroke::new(1.0, Color32::WHITE),
+            );
+            if ui.input_mut(|input| input.consume_key(Modifiers::NONE, Key::Q)) {
+                self.crossings.left_field_corner = match self.crossings.left_field_corner {
+                    Some(_) => None,
+                    None => Some(position),
+                };
+            }
```

Comment:
There is a lot of duplication here.
You could use [`Option::xor(Some(...))`](https://doc.rust-lang.org/std/option/enum.Option.html#method.xor) or better create a function that does this swap for you.
Perhaps this can also do the keypress checking?